airbyte-webapp/src/views/Connector/ConnectorForm/components/Sections/ArraySection.tsx

```diff
@@ -42,15 +42,26 @@ const getItemDescription = (item: Record<string, string>, properties: FormBlock[
 };
 
 export const ArraySection: React.FC<ArraySectionProps> = ({ formField, path, disabled }) => {
-  const [field, , fieldHelper] = useField(path);
-  const [editIndex, setEditIndex] = useState<number>();
+  const [field, , fieldHelper] = useField<Array<Record<string, string>>>(path);
+  const [editIndex, setEditIndex] = useState<number | undefined>();
+  const [itemInEdit, setItemInEdit] = useState<Record<string, string> | undefined>();
```

Comment:
Initially I thought `itemInEdit` contained the new value of the item, and it took me a bit to understand that this is the old value of the item from before it was edited.
Maybe a name like `uneditedItem`, `oldItem`, `prevItem` would make this a bit more clear?

Reply:
Good point, changed